ui/src/components/Pin.tsx

```diff
@@ -59,7 +65,10 @@ export function Pin({
   });
 
   return (
-    <Marker position={[story.latitude, story.longitude]} icon={icon}>
+    <Marker
+      position={[randomCoords(story.latitude), randomCoords(story.longitude)]}
```

Comment:
This doesn't make them unique though, there can still be instances where the positions are the same. Should we guard against those?

Reply:
I kind of don't like this solution at all to be honest since it makes the viewing experience inconsistent, it's just sort of a hacky way to get the clustering working properly. I think in the future we should either
1) let the user place the pin where they would like in a specific city
2) have them enter a general city location, geocode, and then pick somewhere [random ](https://gis.stackexchange.com/questions/25877/generating-random-locations-nearby)within a specific radius? 

We could also make it ordered and just have it fan out directionally from the original coordinate but I don't think that would look as nice. For now, I think since it's a temporary fix, the chances of it not being unique is very, very small?

Reply:
I agree with hanlin on this. We need a fresh design of how coordinates are handled and shown. Current way could work for few stories but when number of stories increase, we could see major trouble. Since the term is finishing I guess this fix could work and next term we could add a proper solution